src/paho/mqtt/client.py

```diff
@@ -524,7 +524,7 @@ def __init__(self, client_id="", clean_session=True, userdata=None, protocol=MQT
         self._host = ""
         self._port = 1883
         self._bind_address = ""
-        self._in_callback = False
+        self._in_callback = threading.Lock()
```

Comment:
I don't know if this is supposed to be a lock. Rather just an indicator that a callback is being executed.